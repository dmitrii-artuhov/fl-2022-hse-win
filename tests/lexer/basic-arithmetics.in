start=🤯expr🤯
🤯expr🤯 👉 🤯expr🤯 🥵+🥵 🤯term🤯 🤌 🤯expr🤯 🥵-🥵 🤯term🤯 🤌 🤯term🤯 🗿
🤯term🤯 👉 🤯term🤯 🥵*🥵 🤯factor🤯 🤌 🤯term🤯 🥵/🥵 🤯factor🤯 🤌 🤯factor🤯 🗿
🤯factor🤯 👉 🥵(🥵 🤯expr🤯 🥵)🥵 🤌 🥵0🥵 🤌 🤯number🤯 🗿
🤯number🤯 👉 🥵0🥵 🤌 🥵1🥵 🤌 🥵2🥵 🤌 🥵3🥵 🤌 🥵4🥵 🤌 🥵5🥵 🤌 🥵6🥵 🤌 🥵7🥵 🤌 🥵8🥵 🤌 🥵9🥵 🗿