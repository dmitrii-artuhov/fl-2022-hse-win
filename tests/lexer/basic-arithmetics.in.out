LexToken(START,'expr',1,0)
LexToken(NON_TERMINAL,'expr',2,13)
LexToken(ARROW,'ðŸ‘‰',2,20)
LexToken(NON_TERMINAL,'expr',2,22)
LexToken(TERMINAL,'+',2,29)
LexToken(NON_TERMINAL,'term',2,33)
LexToken(SEPARATOR,'\U0001f90c',2,40)
LexToken(NON_TERMINAL,'expr',2,42)
LexToken(TERMINAL,'-',2,49)
LexToken(NON_TERMINAL,'term',2,53)
LexToken(SEPARATOR,'\U0001f90c',2,60)
LexToken(NON_TERMINAL,'term',2,62)
LexToken(END,'ðŸ—¿',2,69)
LexToken(NON_TERMINAL,'term',3,71)
LexToken(ARROW,'ðŸ‘‰',3,78)
LexToken(NON_TERMINAL,'term',3,80)
LexToken(TERMINAL,'*',3,87)
LexToken(NON_TERMINAL,'factor',3,91)
LexToken(SEPARATOR,'\U0001f90c',3,100)
LexToken(NON_TERMINAL,'term',3,102)
LexToken(TERMINAL,'/',3,109)
LexToken(NON_TERMINAL,'factor',3,113)
LexToken(SEPARATOR,'\U0001f90c',3,122)
LexToken(NON_TERMINAL,'factor',3,124)
LexToken(END,'ðŸ—¿',3,133)
LexToken(NON_TERMINAL,'factor',4,135)
LexToken(ARROW,'ðŸ‘‰',4,144)
LexToken(TERMINAL,'(',4,146)
LexToken(NON_TERMINAL,'expr',4,150)
LexToken(TERMINAL,')',4,157)
LexToken(SEPARATOR,'\U0001f90c',4,161)
LexToken(TERMINAL,'0',4,163)
LexToken(SEPARATOR,'\U0001f90c',4,167)
LexToken(NON_TERMINAL,'number',4,169)
LexToken(END,'ðŸ—¿',4,178)
LexToken(NON_TERMINAL,'number',5,180)
LexToken(ARROW,'ðŸ‘‰',5,189)
LexToken(TERMINAL,'0',5,191)
LexToken(SEPARATOR,'\U0001f90c',5,195)
LexToken(TERMINAL,'1',5,197)
LexToken(SEPARATOR,'\U0001f90c',5,201)
LexToken(TERMINAL,'2',5,203)
LexToken(SEPARATOR,'\U0001f90c',5,207)
LexToken(TERMINAL,'3',5,209)
LexToken(SEPARATOR,'\U0001f90c',5,213)
LexToken(TERMINAL,'4',5,215)
LexToken(SEPARATOR,'\U0001f90c',5,219)
LexToken(TERMINAL,'5',5,221)
LexToken(SEPARATOR,'\U0001f90c',5,225)
LexToken(TERMINAL,'6',5,227)
LexToken(SEPARATOR,'\U0001f90c',5,231)
LexToken(TERMINAL,'7',5,233)
LexToken(SEPARATOR,'\U0001f90c',5,237)
LexToken(TERMINAL,'8',5,239)
LexToken(SEPARATOR,'\U0001f90c',5,243)
LexToken(TERMINAL,'9',5,245)
LexToken(END,'ðŸ—¿',5,249)
